spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -20,110 +20,126 @@ On Ethereum, there are two types of accounts: externally owned accounts, control
-enum Type {
-  REGISTER,
-  RUNTX
-}
+#### Sending Interface
+The `tryRegisterInterchainAccount` method in the `InterchainAccountModule` interface defines the way to request the creation of an interchain account on a remote chain. The remote chain creates an interchain account using its own account creation logic. Due to the limitation of ordered channels, the recommended way to achieve this when calling `tryRegisterInterchainAccount` is to dynamically bind a new port with the port id set as the address of the owner of the account (if the port is not already bound), invoke `OpenChanInit` via an IBC module which will initiate the handshake process and emit an event signaling to a relayer to generate a new channel between both chains. The remote chain can then create the interchain account in the `ChanOpenTry` callback as part of the channel creation handshake process defined in [ics-4](https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics). Once the `chanOpenAck` callback fires the handshake-originating (sending) chain can assume the account registration is succesful.  
+
+
+The `tryRunTx` method in the`InterchainAccountModule` interface defines the way to create an outgoing packet for a specific chain type. The chain type determines how the IBC account transaction should be constructed and serialised for the recieving chain. The sending side should know in advance how the recieving side expects the incoming IBC packet to be structured. 
+
+#### Recieving Interfacce
+
+`createAccount` defines the way to determine the account's address by using the port & channel id. A newly created interchain account must not conflict with an existing account. Therefore, the host chain (the chain that the account will live on) must keep track of which blockchains have created an interchain account in order to verify the transaction signing authority in `authenticateTx`. `createAccount` should be called in the `chanOpenTry` callback as part of the channel creation handshake process defined in [ics-4](https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics).
 
-interface IBCAccountPacketData {
-  type: Type
-  data: Uint8Array
+`authenticateTx` validates the transaction and checks that the signers in the transaction have the right permissions (are the owners of the account in question).`
+
+`runTx` executes a transaction after the transaction has been successfully authenticated.
+
+### Packet Data
+`InterchainAccountPacketData` contains an array of messages that an interchain account can run and a memo string that is sent to the recieving chain. The example below is defined as a proto encoded message but each chain can encode this differently. This should be clearly defined in the documentation for each chain specific implementation.
+
+```typescript
+message InterchainAccountPacketData  {
+    repeated google.protobuf.Any messages = 1;
+    string memo = 2;
```

Comment:
@ethanfrey What was your idea for using `memo` here? I'm wondering what data is useful to send here.

Reply:
We can drop it. It just came from the standard `sdk.Tx` protobuf type, and figured it might be interesting for documentation. The same way the memo field on transactions doesn't actually do anything but can be observed by clients.

Fine to remove if you don't want it. Figured an optional field wouldn't hurt